From d6d230c956095ad39ebac8a4725ac8e1790f928b Mon Sep 17 00:00:00 2001
From: Nicklas <nicklaslof76@gmail.com>
Date: Sun, 26 Aug 2012 13:23:25 +0200
Subject: [PATCH] Fix the bad precision on Iyokan touch screen

4.1.B.0.431 source broke the precision of the touch screen driver
for Iyokan devices making it very choppy.

Change-Id: I31894aafa66f19c03adf98c2ba5a1c86558bbe1d
---
 arch/arm/mach-msm/touch-iyokan.c        |    4 ++--
 drivers/input/touchscreen/cyttsp_core.c |    4 ++++
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-msm/touch-iyokan.c b/arch/arm/mach-msm/touch-iyokan.c
index 8c4ec23..7aedc88 100644
--- a/arch/arm/mach-msm/touch-iyokan.c
+++ b/arch/arm/mach-msm/touch-iyokan.c
@@ -32,9 +32,9 @@ struct cyttsp_platform_data cyttsp_data = {
 	.use_load_file = 1,
 	.use_force_fw_update = 0,
 	/* activate up groups */
-	.gest_set = CY_GEST_KEEP_ASIS,
+	.gest_set = 0,
 	/* set active distance */
-	.act_dist = CY_ACT_DIST_01,
+	.act_dist = CY_ACT_DIST_03,
 	/* change act_intrvl to customize the Active power state
 	 * scanning/processing refresh interval for Operating mode
 	 */
diff --git a/drivers/input/touchscreen/cyttsp_core.c b/drivers/input/touchscreen/cyttsp_core.c
index 3a50852..468300b 100644
--- a/drivers/input/touchscreen/cyttsp_core.c
+++ b/drivers/input/touchscreen/cyttsp_core.c
@@ -1581,7 +1581,9 @@ again:
 		goto to_bl_mode;
 
 	/* init gesture setup */
+	#ifndef CONFIG_MACH_SEMC_IYOKAN
 	if (ts->platform_data->use_gestures) {
+	#endif
 		u8 gesture_setup;
 
 		DBG(printk(KERN_INFO"%s: Init gesture setup\n", __func__);)
@@ -1603,7 +1605,9 @@ again:
 		if (retval)
 			goto err_gestures;
 		msleep(CY_DELAY_DFLT);
+	#ifndef CONFIG_MACH_SEMC_IYOKAN
 	}
+	#endif
 	goto done;
 
 err_gestures:
-- 
1.7.10.4

